setup.py

```diff
@@ -190,47 +202,25 @@ def gen_packages_items():
 
 
 if __name__ == '__main__':
-    write_version_py()
-    setup(
-        name='mmdet',
-        version=get_version(),
-        description='Open MMLab Detection Toolbox and Benchmark',
-        long_description=readme(),
-        author='OpenMMLab',
-        author_email='chenkaidev@gmail.com',
-        keywords='computer vision, object detection',
-        url='https://github.com/open-mmlab/mmdetection',
-        packages=find_packages(exclude=('configs', 'tools', 'demo')),
-        package_data={'mmdet.ops': ['*/*.so']},
-        classifiers=[
-            'Development Status :: 4 - Beta',
-            'License :: OSI Approved :: Apache Software License',
-            'Operating System :: OS Independent',
-            'Programming Language :: Python :: 3',
-            'Programming Language :: Python :: 3.5',
-            'Programming Language :: Python :: 3.6',
-            'Programming Language :: Python :: 3.7',
-        ],
-        license='Apache License 2.0',
-        setup_requires=parse_requirements('requirements/build.txt'),
-        tests_require=parse_requirements('requirements/tests.txt'),
-        install_requires=parse_requirements('requirements/runtime.txt'),
-        extras_require={
-            'all': parse_requirements('requirements.txt'),
-            'tests': parse_requirements('requirements/tests.txt'),
-            'build': parse_requirements('requirements/build.txt'),
-            'optional': parse_requirements('requirements/optional.txt'),
-        },
-        ext_modules=[
-            make_cuda_ext(
+    cpu_only = False
+    if "--cpu" in sys.argv:
```

Comment:
I suggest using `arg_parser `here

Reply:
i couldn't use normal arg_parser here because later on distutils expect to get specific args and i didn't want to override them